packages/bot-skeleton/src/scratch/hooks/workspace_svg.js

```diff
@@ -79,20 +79,19 @@ Blockly.WorkspaceSvg.prototype.addBlockNode = function (block_node) {
     const { flyout } = DBotStore.instance;
     const block = Blockly.Xml.domToBlock(block_node, flyout.getFlyout().targetWorkspace);
     const top_blocks = this.getTopBlocks(true);
-    const new_block = flyout.getFlyout().createBlock(false, block);
```

Comment:
@farabi-deriv, this will break the functionality of mandatory blocks, as they will not be replaced, leading to duplicated mandatory blocks.

Reply:
@rupato-deriv yes but to fix 2 mandatory blocks, all the rest of the blocks available are getting duplicated due to calling createBlock. I will move the logic of mandatory block from createBlock to this addBlockNode. This way all the blocks will not be duplicated and we will be handling the mandatory blocks.